src/server/gc_worker/gc_manager.rs

```diff
@@ -442,13 +446,27 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine> GcMan
         let mut progress = Some(Key::from_encoded(BEGIN_KEY.to_vec()));
 
         // Records how many region we have GC-ed.
-        let mut processed_regions = 0;
+        let mut scheduled_regions = 0;
+        let task_controller = Arc::new((Mutex::new(0), Condvar::new()));
```

Comment:
seems what you need is https://docs.rs/tokio/latest/tokio/sync/struct.Semaphore.html

Reply:
tokio Semaphore needs the tokio runtime to block,  so with Condvar and Mutex as the alternative implementation. 
@Connor1996